examples/algorithm_progression.py

```diff
@@ -0,0 +1,177 @@
+    return new_source
+
+
+def save_result(source, iteration, params, output_dir, algorithm_name):
+    filename = f'{algorithm_name} {iteration}.tiff'
+    filepath = Path(output_dir, filename)
+    source.write(str(filepath), lossy=False)
+    desc = {'path': filename}
+    for i, v in enumerate(params):
+        desc[VARIABLE_LAYERS[i]] = float(v)
+    return desc
+
+
+def sweep_algorithm(algorithm, input_filename, input_params, output_dir):
+    algorithm_name = algorithm.__name__.replace('_', ' ').title()
+    yaml_dict = {
+        'name': f'{algorithm_name} iterative results',
+        'description': f'{algorithm_name} algorithm performed on {input_filename}',
+        'sources': [{'path': input_filename}],
+    }
+    param_desc = [
+        f'{VARIABLE_LAYERS[i]} = {n}'
+        for i, (n, p) in enumerate(input_params.items())
+        if len(p) > 1 and i < len(VARIABLE_LAYERS)
+    ]
+    if len(param_desc) > 0:
+        yaml_dict['description'] += f', where {", ".join(param_desc)}'
+
+    source = large_image.open(input_filename)
+    param_space_combos = itertools.product(*input_params.values())
+    combos_uniques = itertools.product(
+        *[p for p in input_params.values() if len(p) > 1]
+    )
+
+    print(f'Beginning {len(param_space_combos)} runs on {MAX_THREADS} threads...')
+    with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_THREADS) as executor:
+        for result in executor.map(
+            lambda combo, unique: save_result(
+                apply_algorithm(algorithm, source, combo),
+                *unique,
+                output_dir,
+                algorithm_name,
+            ),
+            param_space_combos,
+            enumerate(combos_uniques),
+        ):
+            yaml_dict['sources'].append(result)
+
+    with open(Path(output_dir, 'results.yml'), 'w') as f:
+        yaml.dump(
+            yaml_dict,
+            f,
+            default_flow_style=False,
+            sort_keys=False,
+        )
+
+
+def get_input_params_from_user(algorithm, sig):
+    print(f'\nProvide parameter space for algorithm {algorithm.__name__}.')
+    print(
+        'Hit enter to accept default for every run, '
+        'provide a single value to replace the default for every run.'
+    )
+    print(
+        'Provide two values separated by a comma '
+        'to specify a range to cover across all runs.'
+    )
+    input_params = {}
+    for param in sig.parameters.values():
+        if param.name != 'data':
+            user_input = input(f'{param.name} ({param.default}): ')
+            if len(user_input.split(',')) == 2:
+                user_range = [float(i.strip()) for i in user_input.split(',')]
+                num_values = int(
+                    input(
+                        f'Enter a number of values to generate in the range {user_range}:'
+                    )
+                )
+                input_params[param.name] = np.linspace(
```

Comment:
Some parameters we will want to range on the semi open-ended interval, e.g. hue-value makes sense to range from [0, 1), while others are conceptually easier to think about in the closed interval, hue_width might range from [0.05, 0.25].  This is a flag to `linspace`.